corehq/apps/auditcare/couch_to_sql.py

```diff
@@ -0,0 +1,201 @@
+        util.set_batch_as_errored(key, last_doc_id, other_doc_count)
+        notify_exception(None, message=message)
+        _soft_assert = soft_assert(to="{}@{}.com".format('aphulera', 'dimagi'), notify_admins=False)
+        _soft_assert(False, message)
+        return
+    logger.info(f"Batch finished: {start_time} - {end_time}")
+    util.set_batch_as_finished(key, count, other_doc_count)
+
+
+def get_couch_key(time):
+    if not time:
+        return
+    return [time.year, time.month, time.day, time.hour, time.minute, time.second]
+
+
+def get_migration_key(start_time, end_time):
+    return get_formatted_datetime_string(start_time) + '_' + get_formatted_datetime_string(end_time)
+
+
+def get_events_from_couch(start_key, end_key, start_doc_id=None):
+    navigation_objects = []
+    access_objects = []
+    records_returned = 0
+    next_start_time = None
+    nav_couch_ids = []
+    access_couch_ids = []
+    other_doc_type_count = 0
+    processed_doc_id = start_doc_id
+    couch_docs = _get_couch_docs(start_key, end_key, start_doc_id)
+    for result in couch_docs:
+        records_returned += 1
+        doc = result["doc"]
+        next_start_time = force_to_datetime(doc.get("event_date"))
+        kwargs = _pick(doc, ["user", "domain", "ip_address", "session_key",
+                            "status_code", "user_agent"])
+        kwargs.update({
+            "event_date": next_start_time,
+            "couch_id": doc["_id"],
+        })
+        processed_doc_id = doc["_id"]
+        if doc["doc_type"] == "NavigationEventAudit":
+            nav_couch_ids.append(doc['_id'])
+            kwargs.update(_pick(doc, ["headers", "status_code", "view", "view_kwargs"]))
+            # Postgres does not play well with control characters in strings
+            # Some crafted URLs can contain these charachters, so replacing them with '' in request_path
+            # https://stackoverflow.com/a/14946355/3537212
+            request_path = re.sub(
+                r'[\x00-\x08\x0b\x0c\x0e-\x1f\x7f-\xff]',
+                lambda match: repr(match.group(0)),
+                doc.get("request_path", "")
+            )
+            path, _, params = request_path.partition("?")
+            kwargs.update({
+                "path": path,
+                "params": params,
+            })
+            navigation_objects.append(NavigationEventAudit(**kwargs))
+        elif doc["doc_type"] == "AccessAudit":
+            access_couch_ids.append(doc['_id'])
+            kwargs.update(_pick(doc, ["http_accept", "trace_id"]))
+            access_type = doc.get('access_type')
+            kwargs.update({
+                "access_type": ACCESS_LOOKUP.get(doc.get("access_type")),
+                "path": doc.get("path_info"),
+            })
+            if access_type == "logout":
+                kwargs.update({"path": "accounts/logout"})
+            access_objects.append(AccessAudit(**kwargs))
+        else:
+            other_doc_type_count += 1
```

Comment:
I think this branch should check the doc type against the set of expected doc types to be ignored and throw an exception if a type is encountered that we do not expect. Could be as simple as
```py
assert doc["doc_type"] in IGNORED_DOC_TYPES, doc
```
Of course `IGNORED_DOC_TYPES = {"IgnoredType", ...}` will need to be defined above.

Reply:
Cool.